dotCMS/src/main/java/com/dotmarketing/portlets/rules/business/RulesEngine.java

```diff
@@ -68,6 +73,75 @@ public static void fireRules(HttpServletRequest req, HttpServletResponse res, Ru
 		fireRules(req, res, host, fireOn);
 	}
 	
+	
+	
+	
+
+    public static void fireRules(final HttpServletRequest request, final HttpServletResponse response) {
+        
+        Optional<Visitor> visitor = APILocator.getVisitorAPI().getVisitor(request);
+
+        boolean newVisitor = false;
+        boolean newVisit = false;
+
+        /*
+         * JIRA http://jira.dotmarketing.net/browse/DOTCMS-4659 //Set long lived cookie
+         * regardless of who this is
+         */
+        String _dotCMSID =
+                UtilMethods.getCookieValue(request.getCookies(), com.dotmarketing.util.WebKeys.LONG_LIVED_DOTCMS_ID_COOKIE);
+
+        if (!UtilMethods.isSet(_dotCMSID)) {
+            // create unique generator engine
+            Cookie idCookie = CookieUtil.createCookie();
+            _dotCMSID = idCookie.getValue();
+            response.addCookie(idCookie);
+            newVisitor = true;
+
+            if (visitor.isPresent()) {
+                visitor.get()
+                    .setDmid(UUID.fromString(_dotCMSID));
+            }
+
+        }
+
+        String _oncePerVisitCookie = UtilMethods.getCookieValue(request.getCookies(), WebKeys.ONCE_PER_VISIT_COOKIE);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00117)